src/NzbDrone.Core/Indexers/Definitions/Xthor/Xthor.cs

```diff
@@ -0,0 +1,470 @@
+                    // Passkey not found
+                    throw new Exception("Passkey not found in tracker's database");
+                case 2:
+                    // No results
+                    break;
+                case 3:
+                    // Power Saver
+                    break;
+                case 4:
+                    // DDOS Attack, API disabled
+                    throw new Exception("Tracker is under DDOS attack, API disabled");
+                case 8:
+                    // AntiSpam Protection
+                    throw new Exception("Triggered AntiSpam Protection, please delay your requests !");
+                default:
+                    // Unknown state
+                    throw new Exception("Unknown state, aborting querying");
+            }
+        }
+
+        public Action<IDictionary<string, string>, DateTime?> CookiesUpdater { get; set; }
+    }
+
+    public class XthorSettingsValidator : AbstractValidator<XthorSettings>
+    {
+    }
+
+    public class XthorSettings : IIndexerSettings
+    {
+        private static readonly XthorSettingsValidator Validator = new XthorSettingsValidator();
+
+        public XthorSettings()
+        {
+            BaseUrl = "https://api.xthor.tk/";
+            Passkey = "";
+            FreeleechOnly = false;
+            Accent = 0;
+            NeedMultiReplacement = false;
+            MultiReplacement = "";
+            SubReplacement = "";
+            EnhancedAnime = true;
+            ByPassPageForTmDbid = true;
+            MaxPages = 1;
+        }
+
+        [FieldDefinition(1,
+            Label = "Url",
+            Type = FieldType.Select,
+            SelectOptionsProviderAction = "getUrls",
+            HelpText = "Site links")]
+        public string BaseUrl { get; set; }
+
+        [FieldDefinition(2,
+            Label = "Passkey",
+            Privacy = PrivacyLevel.Password,
+            Type = FieldType.Password,
+            HelpText = "Passkey to use for requests. (see https://xthor.tk/api_doc.php)")]
+        public string Passkey { get; set; }
+
+        [FieldDefinition(2,
+            Label = "Freeleech only",
+            Privacy = PrivacyLevel.Normal,
+            Type = FieldType.Checkbox,
+            HelpText =
+                "If you want to discover only freeleech torrents to not impact your ratio, check the related box. So only torrents marked as freeleech will be returned instead of all.")]
+        public bool FreeleechOnly { get; set; }
+
+        [FieldDefinition(3,
+            Label = "Specific language",
+            Type = FieldType.Select,
+            SelectOptions = typeof(XthorAccent),
+            HelpText = "You can scope your searches with a specific language / accent.")]
+
+        public int Accent { get; set; }
+
+        public NzbDroneValidationResult Validate()
+        {
+            return new NzbDroneValidationResult(Validator.Validate(this));
+        }
+
+        [FieldDefinition(4,
```

Comment:
single line as well;  should this be advanced?

Reply:
Fixed, not advanced for me (it's a replacement that anybody can use because of the naming on this tracker)